proposals/1125_dynamic-supervisor-oidc-clients/README.md

```diff
@@ -0,0 +1,507 @@
+  memberships or not, to allow the admin to exclude this potentially information for clients which do not need it.
+- Support a web UI based LDAP/ActiveDirectory login screen. This is needed to avoid having webapps handle the user's
+  password, which must only be seen by the Supervisor and the LDAP server. However, the details of this item have been
+  split out to a separate proposal document.
+- Client secrets must be stored encrypted or hashed, not in plain text.
+- Creation of client credentials on the operator's behalf - the server must generate any secrets.
+- The operator must be able to initiate manual rotation of client credentials.
+- Documentation describing the token exchanges a webapp backend must perform to interact with the Kubernetes API.
+
+Non-goals for this proposal:
+
+- Pinniped's scope is to provide authentication for cluster users. Providing authentication for arbitrary users to
+  arbitrary webapps is out of scope. The only proposed use case is providing the exact same identities that are provided
+  by using Pinniped's `kubectl` integration, which are the developers/devops/admin users of the cluster.
+- Supporting any OAuth/OIDC flow other
+  than [OIDC authorization code flow](https://openid.net/specs/openid-connect-core-1_0.html#CodeFlowAuth).
+- Implementing any self-service client registration API. Clients will be registered by the Pinniped admin user.
+- Implementing a consent screen. This would be clearly valuable but will be left as a potential future enhancement in
+  the interest of keeping the first draft of this feature smaller.
+- Orchestration of token exchanges on behalf of the client. Webapps which want to make calls to the Kubernetes API of
+  clusters acting as the authenticated user will need to perform the rest of the token and credential exchange flow that
+  it currently implemented by the Pinniped CLI. Providing some kind of component or library to assist webapp developers
+  with these steps might be valuable but will be left as a potential future enhancement.
+- Supporting JWT-based client auth as described in [RFC 7523](https://datatracker.ietf.org/doc/html/rfc7523).  For now
+  client secret basic auth will be used.  This is left as a potential future enhancement.
+- Supporting public clients.
+- Supporting any policy around which users an OAuth client can interact with.  This is left as a potential future
+  enhancement.
+
+### Specification / How it Solves the Use Cases
+
+This document proposes supporting a new Custom Resource Definition (CRD) for the Pinniped Supervisor which allows the
+admin to create, update, and delete OIDC clients for the Supervisor.
+
+#### API Changes
+
+##### Configuring clients
+
+An example of the new CRD to define a client:
+
+```yaml
+apiVersion: oauth.supervisor.pinniped.dev/v1alpha1
+kind: OIDCClient
+metadata:
+  name: client.oauth.pinniped.dev-my-webapp-client
+  namespace: pinniped-supervisor
```

Comment:
Do users need to be able to create the `OIDCClient` in the `pinniped-supervisor` namespace, or can they be created in the namespace in which the user has the relevant RBAC (EDIT: I see below that you've outlined it has to be in the same namespace as the supervisor as it is configuration for the supervisor)?

Reply:
Well, might be harder to control delegated permissions from having read/write access on each other's (without something more granular like open policy agent) - which is why I would have thought namespaces could be useful if there was any thought of delegation. But ok, if it's admin-only, makes sense.